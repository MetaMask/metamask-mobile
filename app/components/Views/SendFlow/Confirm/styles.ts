import { fontStyles } from '../../../../styles/common';
import { StyleSheet } from 'react-native';

const createStyles = (colors: any) =>
  StyleSheet.create({
    wrapper: {
      flex: 1,
      backgroundColor: colors.background.default,
    },
    inputWrapper: {
      flex: 0,
      borderBottomWidth: 1,
      borderBottomColor: colors.border.default,
      paddingHorizontal: 8,
    },
    amountWrapper: {
      flexDirection: 'column',
      margin: 24,
    },
    textAmountLabel: {
      ...fontStyles.normal,
      fontSize: 14,
      textAlign: 'center',
      color: colors.text.alternative,
      textTransform: 'uppercase',
      marginVertical: 3,
    },
    textAmount: {
      ...fontStyles.light,
      color: colors.text.default,
      fontSize: 44,
      textAlign: 'center',
    },
    buttonNext: {
      flex: 1,
      marginHorizontal: 24,
      alignSelf: 'flex-end',
    },
    buttonNextWrapper: {
      flexDirection: 'row',
      alignItems: 'flex-end',
      marginBottom: 16,
    },
    actionTouchable: {
      padding: 12,
    },
    actionText: {
      ...fontStyles.normal,
      color: colors.primary.default,
      fontSize: 14,
      alignSelf: 'center',
    },
    actionsWrapper: {
      margin: 24,
    },
    CollectibleMediaWrapper: {
      flexDirection: 'column',
      alignItems: 'center',
      margin: 16,
    },
    collectibleName: {
      ...fontStyles.normal,
      fontSize: 18,
      color: colors.text.default,
      textAlign: 'center',
    },
    collectibleTokenId: {
      ...fontStyles.normal,
      fontSize: 12,
      color: colors.text.alternative,
      marginTop: 8,
      textAlign: 'center',
    },
    CollectibleMedia: {
      height: 120,
      width: 120,
    },
    qrCode: {
      marginBottom: 16,
      paddingHorizontal: 36,
      paddingBottom: 24,
      paddingTop: 16,
      backgroundColor: colors.background.default,
      borderRadius: 8,
      width: '100%',
    },
    hexDataWrapper: {
      padding: 10,
      alignItems: 'center',
      borderRadius: 10,
      backgroundColor: colors.background.default,
    },
    addressTitle: {
      ...fontStyles.bold,
      color: colors.text.default,
      alignItems: 'center',
      justifyContent: 'center',
      textAlign: 'center',
      fontSize: 16,
      marginBottom: 16,
    },
    hexDataClose: {
      zIndex: 999,
      position: 'absolute',
      top: 12,
      right: 20,
    },
    hexDataText: {
      textAlign: 'justify',
    },
    bottomModal: {
      justifyContent: 'flex-end',
      margin: 0,
    },
    keyboardAwareWrapper: {
      flex: 1,
      justifyContent: 'flex-end',
    },
    errorWrapper: {
      marginHorizontal: 24,
      marginTop: 12,
      paddingHorizontal: 10,
      paddingVertical: 8,
      backgroundColor: colors.error.muted,
      borderColor: colors.error.default,
      borderRadius: 8,
      borderWidth: 1,
      justifyContent: 'center',
      alignItems: 'center',
    },
    error: {
      color: colors.text.default,
      fontSize: 12,
      lineHeight: 16,
      ...fontStyles.normal,
      textAlign: 'center',
    },
    underline: {
      textDecorationLine: 'underline',
      ...fontStyles.bold,
    },
    text: {
      lineHeight: 20,
      color: colors.text.default,
    },
  });

export default createStyles;
