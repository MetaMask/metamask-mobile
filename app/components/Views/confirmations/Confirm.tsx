// This is an temporary file to export the Confirm component for outside of confirmations folder
// And will be removed immediately after folder restructuring is done
export * from './components/confirm';
