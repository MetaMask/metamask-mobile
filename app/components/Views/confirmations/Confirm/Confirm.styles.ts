import { StyleSheet } from 'react-native';

import Device from '../../../../util/device';
import { Theme } from '../../../../util/theme/models';

const styleSheet = (params: { theme: Theme }) => {
  const { theme } = params;

  return StyleSheet.create({
    container: {
      backgroundColor: theme.colors.background.alternative,
      paddingHorizontal: 16,
      paddingVertical: 24,
      minHeight: '60%',
      borderTopLeftRadius: 20,
      borderTopRightRadius: 20,
      paddingBottom: Device.isIphoneX() ? 20 : 0,
      alignItems: 'center',
      justifyContent: 'space-between'
    },
  });
};

export default styleSheet;
