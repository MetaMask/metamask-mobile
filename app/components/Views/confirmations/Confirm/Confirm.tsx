import React from 'react';
import { Text, View } from 'react-native';

import BottomModal from '../../../../components/UI/BottomModal';
import { useTheme } from '../../../../util/theme';
import useRedesignEnabled from '../hooks/useRedesignEnabled';
import createStyles from './style';

const Confirm = () => {
  const { colors } = useTheme();
  const { isRedesignedEnabled } = useRedesignEnabled();

  if (!isRedesignedEnabled) {
    return null;
  }

  const styles = createStyles(colors);

  return (
    <BottomModal>
      <View style={styles.container}>
        <Text>TODO</Text>
      </View>
    </BottomModal>
  );
};

export default Confirm;
