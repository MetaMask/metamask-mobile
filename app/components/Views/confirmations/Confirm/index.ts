export { Confirm } from './Confirm';
export { ConfirmRoot } from './ConfirmRoot';
