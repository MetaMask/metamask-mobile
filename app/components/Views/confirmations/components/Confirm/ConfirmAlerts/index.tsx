export { default as ConfirmAlerts } from './ConfirmAlerts';
