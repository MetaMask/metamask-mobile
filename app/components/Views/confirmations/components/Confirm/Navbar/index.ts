export { getNavbar as getStakingNavbar } from './Navbar';
