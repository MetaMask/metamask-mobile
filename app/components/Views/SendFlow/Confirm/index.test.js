import React from 'react';
import { shallow } from 'enzyme';
import Confirm from './';
import configureMockStore from 'redux-mock-store';

describe('Confirm', () => {
	const mockStore = configureMockStore();
	it('should render correctly', () => {
		const initialState = {
			engine: {
				backgroundState: {
					NetworkController: {
						network: '1',
						provider: {
							ticker: 'ETH',
							type: 'mainnet'
						}
					},
					AccountTrackerController: {
						accounts: { '0x2': { balance: '0' } }
					},
					TransactionController: {
						transactions: []
					},
					TokenRatesController: {
						contractExchangeRates: {}
					},
					CurrencyRateController: {
						currentCurrency: 'USD',
						conversionRate: 1
					},
					TokenBalancesController: {
						contractBalances: {}
					},
					PreferencesController: {
						identities: {}
					},
					KeyringController: {
						keyrings: {}
					}
				}
			},
			settings: {
				showHexData: true
			},
			transaction: {
				selectedAsset: {},
				transaction: {
					from: '0x1',
					to: '0x2'
				}
			}
		};
		const wrapper = shallow(<Confirm />, {
			context: { store: mockStore(initialState) }
		});
		expect(wrapper.dive()).toMatchSnapshot();
	});
});
