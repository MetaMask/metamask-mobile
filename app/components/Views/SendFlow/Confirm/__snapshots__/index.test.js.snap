// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm should render correctly 1`] = `
<ForwardRef(SafeAreaView)
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
  testID="txn-confirm-screen"
>
  <View
    style={
      Object {
        "borderBottomColor": "#D8D8D8",
        "borderBottomWidth": 1,
        "flex": 0,
        "paddingHorizontal": 8,
      }
    }
  >
    <AddressFrom
      fromAccountAddress="0x1"
      onPressIcon={null}
    />
    <AddressTo
      addressToReady={true}
      toSelectedAddress=""
    />
  </View>
  <ScrollView
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "column",
          "margin": 24,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#6a737d",
            "fontFamily": "EuclidCircularB-Regular",
            "fontSize": 14,
            "fontWeight": "400",
            "marginVertical": 3,
            "textAlign": "center",
            "textTransform": "uppercase",
          }
        }
      >
        Amount
      </Text>
      <Text
        style={
          Object {
            "color": "#24292E",
            "fontFamily": "EuclidCircularB-Regular",
            "fontSize": 44,
            "fontWeight": "300",
            "textAlign": "center",
          }
        }
        testID="confirm-txn-amount"
      />
      <Text
        style={
          Object {
            "color": "#6a737d",
            "fontFamily": "EuclidCircularB-Regular",
            "fontSize": 14,
            "fontWeight": "400",
            "marginVertical": 3,
            "textAlign": "center",
            "textTransform": "uppercase",
          }
        }
      />
    </View>
    <TransactionReviewFeeCard
      edit={[Function]}
      fiat=""
      gasEstimationReady={false}
      over={false}
      totalFiat={<Text />}
      totalGasFiat=""
      totalValue={<Text />}
      transactionValue=""
    />
    <View
      style={
        Object {
          "margin": 24,
        }
      }
    >
      <TouchableOpacity
        onPress={[Function]}
        style={
          Object {
            "padding": 12,
          }
        }
      >
        <Text
          style={
            Object {
              "alignSelf": "center",
              "color": "#037dd6",
              "fontFamily": "EuclidCircularB-Regular",
              "fontSize": 14,
              "fontWeight": "400",
            }
          }
        >
          Hex Data
        </Text>
      </TouchableOpacity>
    </View>
  </ScrollView>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "flexDirection": "row",
        "marginBottom": 16,
      }
    }
  >
    <StyledButton
      containerStyle={
        Object {
          "alignSelf": "flex-end",
          "flex": 1,
          "marginHorizontal": 24,
        }
      }
      disabled={true}
      disabledContainerStyle={
        Object {
          "opacity": 0.6,
        }
      }
      onPress={[Function]}
      styleDisabled={
        Object {
          "opacity": 0.6,
        }
      }
      testID="txn-confirm-send-button"
      type="confirm"
    >
      Send
    </StyledButton>
  </View>
  <ReactNativeModal
    animationIn="slideInUp"
    animationInTiming={300}
    animationOut="slideOutDown"
    animationOutTiming={300}
    avoidKeyboard={false}
    backdropColor="black"
    backdropOpacity={0.7}
    backdropTransitionInTiming={300}
    backdropTransitionOutTiming={300}
    coverScreen={true}
    customBackdrop={null}
    deviceHeight={null}
    deviceWidth={null}
    hasBackdrop={true}
    hideModalContentWhileAnimating={false}
    isVisible={false}
    onBackButtonPress={[Function]}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalShow={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onSwipeComplete={[Function]}
    propagateSwipe={true}
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "justifyContent": "flex-end",
        "margin": 0,
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeDirection="down"
    swipeThreshold={100}
    useNativeDriver={false}
  >
    <Connect(AccountList)
      enableAccountsAddition={false}
      getBalanceError={[Function]}
      identities={Object {}}
      keyrings={
        Array [
          Object {
            "accounts": Array [
              "0x",
            ],
            "type": "HD Key Tree",
          },
        ]
      }
      onAccountChange={[Function]}
      selectedAddress="0x1"
      ticker="ETH"
    />
  </ReactNativeModal>
  <ReactNativeModal
    animationIn="slideInUp"
    animationInTiming={300}
    animationOut="slideOutDown"
    animationOutTiming={300}
    avoidKeyboard={false}
    backdropColor="black"
    backdropOpacity={0.7}
    backdropTransitionInTiming={300}
    backdropTransitionOutTiming={300}
    coverScreen={true}
    customBackdrop={null}
    deviceHeight={null}
    deviceWidth={null}
    hasBackdrop={true}
    hideModalContentWhileAnimating={false}
    isVisible={false}
    onBackButtonPress={[Function]}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalShow={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onSwipeComplete={[Function]}
    propagateSwipe={true}
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeDirection="down"
    swipeThreshold={100}
    useNativeDriver={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "padding": 10,
        }
      }
    >
      <TouchableOpacity
        onPress={[Function]}
        style={
          Object {
            "position": "absolute",
            "right": 20,
            "top": 12,
            "zIndex": 999,
          }
        }
      >
        <Icon
          allowFontScaling={false}
          color="#24292E"
          name="ios-close"
          size={28}
        />
      </TouchableOpacity>
      <View
        style={
          Object {
            "backgroundColor": "#f2f3f4",
            "borderRadius": 8,
            "marginBottom": 16,
            "paddingBottom": 24,
            "paddingHorizontal": 36,
            "paddingTop": 16,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "alignItems": "center",
              "color": "#24292E",
              "fontFamily": "EuclidCircularB-Bold",
              "fontSize": 16,
              "fontWeight": "600",
              "justifyContent": "center",
              "marginBottom": 16,
              "textAlign": "center",
            }
          }
        >
          Hex Data
        </Text>
        <Text
          style={
            Object {
              "textAlign": "justify",
            }
          }
        >
          0x
        </Text>
      </View>
    </View>
  </ReactNativeModal>
</ForwardRef(SafeAreaView)>
`;
